exdir/core/attribute.py

```diff
@@ -22,95 +22,105 @@ def __init__(self, parent, mode, io_mode, path=None, plugin_manager=None):
+            return attrs
 
     def __setitem__(self, name, value):
-        meta_data = self._open_or_create()
-
-        # if isinstance(name, np.integer):
-        #     key = int(name)
-        # else:
-        #     key = name
+        attrs = self._open_or_create()
         key = name
-
-        sub_meta_data = meta_data
+        sub_attrs = attrs
+        
         for i in self.path:
-            sub_meta_data = sub_meta_data[i]
-        sub_meta_data[key] = value
+            sub_attrs = sub_attrs[i]
+        sub_attrs[key] = value
 
-        self._set_data(meta_data)
+        self._set_data(attrs)
 
     def __contains__(self, name):
-        meta_data = self._open_or_create()
+        attrs = self._open_or_create()
         for i in self.path:
-            meta_data = meta_data[i]
-        return name in meta_data
+            attrs = attrs[i]
+        return name in attrs
 
     def keys(self):
-        meta_data = self._open_or_create()
+        attrs = self._open_or_create()
         for i in self.path:
-            meta_data = meta_data[i]
-        return meta_data.keys()
+            attrs = attrs[i]
+        return attrs.keys()
 
     def to_dict(self):
-        meta_data = self._open_or_create()
+        attrs = self._open_or_create()
         for i in self.path:  # TODO check if this is necesary
-            meta_data = meta_data[i]
+            attrs = attrs[i]
 
         for plugin in self.plugin_manager.attribute_plugins.read_order:
-            meta_data = plugin.prepare_read(meta_data)
+            attrs = plugin.prepare_read(attrs)
 
-        return meta_data
+        return attrs
 
     def items(self):
-        meta_data = self._open_or_create()
+        attrs = self._open_or_create()
         for i in self.path:
-            meta_data = meta_data[i]
-        return meta_data.items()
+            attrs = attrs[i]
+        return attrs.items()
 
     def values(self):
-        meta_data = self._open_or_create()
+        attrs = self._open_or_create()
         for i in self.path:
-            meta_data = meta_data[i]
-        return meta_data.values()
+            attrs = attrs[i]
+        return attrs.values()
 
-    def _set_data(self, meta_data):
+    def _set_data(self, attrs):
         if self.io_mode == exob.Object.OpenMode.READ_ONLY:
             raise IOError("Cannot write in read only ("r") mode")
 
+        meta = {}
+        attribute_data = exdir.plugin_interface.AttributeData(attrs=attrs,
+                                                              meta=meta)
+
         for plugin in self.plugin_manager.attribute_plugins.write_order:
-            meta_data = plugin.prepare_write(meta_data)
+            attribute_data = plugin.prepare_write(attribute_data)
+
+            if "required" in attribute_data.meta and attribute_data.meta["required"] is True:
+                meta[plugin._plugin_module.name] = attribute_data.meta
+
+        # QUESTION: Should we add plugin meta data to parent.meta here?
```

Comment:
We need to decide how to include meta information (e.g. `required: True`) about an attribute plugin. One option is to update `parent.meta["plugins"]`. What do you think @dragly?

Reply:
Yes, I think so. I'll look into it now.